src/main/java/org/apache/commons/lang3/StringUtils.java

```diff
@@ -8026,34 +8026,31 @@ private static boolean endsWith(final CharSequence str, final CharSequence suffi
      * @since 3.0
      */
     public static String normalizeSpace(final String str) {
-        // LANG-1020: Improved performance significantly by normalizing manually instead of using regex
-        // See https://github.com/librucha/commons-lang-normalizespaces-benchmark for performance test
         if (isEmpty(str)) {
             return str;
         }
+        if (isBlank(str)){
+           return EMPTY;
+        }
+        
         final int size = str.length();
         final char[] newChars = new char[size];
         int count = 0;
         int whitespacesCount = 0;
-        boolean startWhitespaces = true;
         for (int i = 0; i < size; i++) {
             char actualChar = str.charAt(i);
-            boolean isWhitespace = Character.isWhitespace(actualChar);
+            boolean isWhitespace = Character.isWhitespace(actualChar) || actualChar == '\u00A0';
```

Comment:
Noting that the javadoc doesn't indicate that nbsp is escaped. I'm +1 to also escaping nbsp, but javadoc should be updated.